ceph.spec.in

```diff
@@ -556,6 +557,9 @@ Requires:	libradosstriper1 = %{_epoch_prefix}%{version}-%{release}
 %if 0%{?suse_version}
 Requires(pre):	pwdutils
 %endif
+%if 0%{with ubbd}
+Requires:	ubbd
```

Comment:
I'm not sure this would work given that no distribution ships `ubbd` package today.  This is a client-side thing so we can't easily work around that by putting it in our container either.  Unlike with `-t nbd` where we spawn `rbd-nbd` process, basically limiting the interaction to the CLI, with `-t ubbd` as implemented here we need `libubbd.so` to be installed on the system.

To (temporarily) remove the client-side shared library dependency, perhaps `src/tools/rbd/action/Ubbd.cc` could be changed to spawn [ubbdadm](https://github.com/DataTravelGuide/ubbd/tree/master/ubbdadm) process similar to `src/tools/rbd/action/Nbd.cc`?  This is worse from the design perspective but makes integration with something that isn't universally available much easier -- we could even enable `WITH_RBD_UBBD` by default then.

Or perhaps we could skip daily teuthology testing and just rely on your organization's CI for now, until this
> @danieldin95 is working to contribute deb package of ubbd to debian, that's on the way.

work lands?  Practically this would mean going back to the original version of this PR, with just `src/tools/rbd`, `src/test/cli/rbd` and doc changes.

@yangdongsheng What is your take?

Reply:
@idryomov 
Thanx for your reply.

I would like second way, let's skipping teuthology testing firstly (but ubbd-tests have a same test script for it).
Start from src/tools/rbd only and enable package related work in ceph later, that sounds fair enough to me.

Reply:
> Start from src/tools/rbd only and enable package related work later, that sounds fair enough to me.

Feel free to go ahead and amend the PR accordingly but note that we won't be able to enable `WITH_RBD_UBBD` by default until then.

Reply:
sure thanx